mock/py/mockbuild/backend.py

```diff
@@ -240,6 +225,61 @@ def installSpecDeps(self, spec_file):
         finally:
             self.uid_manager.restorePrivs()
 
+    # when python 3.9 becomes standard, this can be replaced by string.removeprefix()
+    @classmethod
+    def _remove_prefix(cls, intext, prefix):
+        if intext.startswith(prefix):
+            return intext[len(prefix):]
+        return intext
+
+    @traceLog()
+    def install_external_deps(self, deps):
+        """Install dependencies using native library manager"""
+        prefix = 'external:pypi:'
+        pypi_deps = [self._remove_prefix(i, prefix) for i in deps if i.startswith(prefix)]
+        deps = [i for i in deps if not i.startswith(prefix)]
+        self.buildroot.root_log.info('Installing dependencies to satisfy external:*')
+        if pypi_deps:
+            self.install_external_deps_pypi(pypi_deps)
+
+        if deps:
+            raise ExternalDepsError("Unknown external dependencies: {}".format(', ', deps))
+
+    @traceLog()
+    def install_external_deps_pypi(self, deps):
+        """ deps is list of python modules. Without the prefix. """
+        self.buildroot.install_as_root('/usr/bin/pip3', 'python3-setuptools')
+        command = ['pip3', "install"] + deps
+        try:
+            self.uid_manager.becomeUser(0, 0)
+            self.buildroot.doChroot(
+                command, shell=False,
+                nspawn_args=self._get_nspawn_args(),
+                returnOutput=True, unshare_net=False,
+                returnStderr=True, raiseExc=True)
+        finally:
+            self.uid_manager.restorePrivs()
+        self.install_fake_rpm('pypi', deps)
+
+    @traceLog()
+    def install_fake_rpm(self, external_type, deps):
+        """ Create and install fake packages using create-fake-rpm """
+        list_of_packages = []
+        self.buildroot.install_as_root('create-fake-rpm')
+        for dep in deps:
+            command = ['/usr/bin/create-fake-rpm', '--print-result',
+                       '--build',
+                       'external-{0}-{1}'.format(external_type, dep),
+                       'external:{0}:{1}'.format(external_type, dep)]
+            output = self.bootstrap_buildroot.doChroot(
+                command, shell=False,
+                returnOutput=True,
+                printOutput=True,
+                returnStderr=False, raiseExc=True)
+            package = output.split()[0]
+            list_of_packages.append(package)
```

Comment:
There's new Suggests, while we run it eventually in bootstrap ... I don't know, perhaps you eventually want to build the fake RPM directly in the target chroot.